src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultStates = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    } else if (action.type === 'clear') {
+      newState = {};
+    }
```

Comment:
[CODE STYLE]: Use a switch statement instead of multiple if/else if statements when you have a limited amount of conditions to make the code cleaner and more readable. Refactor the if/else if conditions into a switch statement with cases for 'addProperties', 'removeProperties', and 'clear'.